From bd2ab9590af8d68d6a2dd034940fb641129c11a9 Mon Sep 17 00:00:00 2001
From: Istio Automation <istio-testing-bot@google.com>
Date: Fri, 17 Dec 2021 07:11:41 -0800
Subject: [release-1.12] Update gateway service to actually use service
 annotations (#36397)

* Update service.yaml

else the chart is barely usable in any public clouds

* Update manifests/charts/gateway/templates/service.yaml

Co-authored-by: John Howard <howardjohn@google.com>

Co-authored-by: Gleb Boushev <spam@4c74356b41.com>
Co-authored-by: John Howard <howardjohn@google.com>
---
 manifests/charts/gateway/templates/service.yaml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/manifests/charts/gateway/templates/service.yaml b/manifests/charts/gateway/templates/service.yaml
index 7b5253b9d6..068d9d3d40 100644
--- a/manifests/charts/gateway/templates/service.yaml
+++ b/manifests/charts/gateway/templates/service.yaml
@@ -10,7 +10,7 @@ metadata:
     topology.istio.io/network: "{{.}}"
     {{- end }}
   annotations:
-    {{- .Values.annotations | toYaml | nindent 4 }}
+    {{- merge (deepCopy .Values.service.annotations) .Values.annotations | toYaml | nindent 4 }}
 spec:
 {{- with .Values.service.loadBalancerIP }}
   loadBalancerIP: "{{ . }}"
@@ -42,4 +42,4 @@ spec:
 {{- end }}
   selector:
     {{- include "gateway.selectorLabels" . | nindent 4 }}
-{{- end }}
\ No newline at end of file
+{{- end }}
-- 
2.35.3

